### PR TITLE
Make note about HTML-encoding Query Strings in Search Docs

### LINKED ISSUE #819: Search Docs don't specify HTML-encoding query strings
### What article on docs.github.com is affected?

https://docs.github.com/en/free-pro-team@latest/rest/reference/search#constructing-a-search-query

### What part(s) of the article would you like to see updated?

As the docs stand, there is no mention of HTML-encoding the query strings. Instead, the current examples are pseudo-encoded. It would be helpful to specify that users need to encode their query strings before sending them, using their language's preferred HTML-encoder.

### PR DESCRIPTION
**Pull request by wesmangum**
### Why:

Before, the examples of constructing a query string in this document were a combination of unencoded and encoded HTML strings. Now, the examples explicitly state that users should construct their query strings using the HTML-encoder of their choice.

Issue: https://github.com/github/docs/issues/819

### What's being changed:

The "Search Code" docs are updated with a note about HTML-encoding query strings, along with a JavaScript example.

### Check off the following:
- [ ] All of the tests are passing.
- [ ] I have reviewed my changes in staging.
- [x] For content changes, I have reviewed the [localization checklist](https://github.com/github/docs/blob/main/contributing/localization-checklist.md)
- [x] For content changes, I have reviewed the [Content style guide for GitHub Docs](https://github.com/github/docs/blob/main/contributing/content-style-guide.md).

### CODE REVIEW

**Comment by janiceilene** [comment]
Thanks for opening a PR @wesmangum! I'll get this triaged for review 🌟

**Review by husinano — APPROVED** [review]

File: content/rest/reference/search.md
Comment (rachmari) [inline content/rest/reference/search.md:49]: This just formats the note like other notes in our docs:

```suggestion
**Note:** Be sure to use your language's preferred HTML-encoder to construct your query strings. For example:
```

**Review by rachmari — COMMENTED** [review]
@wesmangum thanks for this update! 🙌 This looks great. I just had one small request to update the format of the note. Once that is updated, I'll get this merged! 🚀 🚢

**Comment by janiceilene** [comment]
I committed @rachmari's ✨ suggestion and updated the branch! I'll merge this once tests are passing 🍏

**Review by janiceilene — APPROVED** [review]

**Pull request merged**
